cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -380,6 +411,23 @@ Result<S3Options> S3Options::FromUri(const std::string& uri_string,
   return FromUri(uri, out_path);
 }
 
+std::shared_ptr<S3RetryStrategy> GetS3RetryStrategy(const std::string& name,
+                                                    long retry_attempts) {
+  if (name == "aws_standard") {
+    auto aws_standard = std::make_shared<AwsRetryStrategy>(
+        std::make_shared<Aws::Client::StandardRetryStrategy>(retry_attempts));
+    return std::static_pointer_cast<S3RetryStrategy>(aws_standard);
```

Comment:
I don't think the `static_pointer_cast` is required since implicit conversion should be available.